docs/source/guide/configuration.rst

```diff
@@ -17,100 +20,319 @@ default region::
+    you want to change this location.
+
+``AWS_CA_BUNDLE``
+    The path to a custom certificate bundle to use when establishing
+    SSL/TLS connections.  Boto3 includes a bundled CA bundle it will
+    use by default, but you can set this environment variable to use
+    a different CA bundle.
+
+``AWS_METADATA_SERVICE_TIMEOUT``
+    The number of seconds before a connection to the instance metadata
+    service should time out.  When attempting to retrieve credentials
+    on an EC2 instance that has been configured with an IAM role,
+    a connection to the instance metadata service will time out after
+    1 second by default.  If you know you are running on an EC2 instance
+    with an IAM role configured, you can increase this value if needed.
+
+``AWS_METADATA_SERVICE_NUM_ATTEMPTS``
+    When attempting to retrieve credentials on an EC2 instance that has
+    been configured with an IAM role, boto3 will only make one attempt
+    to retrieve credentials from the instance metadata service before
+    giving up.  If you know your code will be running on an EC2 instance,
+    you can increase this value to make boto3 retry multiple times
+    before giving up.
+
+
+Configuration File
+~~~~~~~~~~~~~~~~~~
+
+Boto3 will also search the ``~/.aws/config`` file when looking for
+configuration values.  You can change the location of this file by
+setting the ``AWS_CONFIG_FILE`` environment variable.
+
+This file is an INI formatted file that contains at least one
+section: ``[default]``.  You can create multiple profiles (logical
+groups of configuration) by creating sections named ``[profile profile-name]``.
+If your profile name has spaces, you'll need to surround this value in quotes:
+``[profile "my profile name"]``.  Below are all the config variables supported
+in the ``~/.aws/config`` file:
 
-It also supports profiles, but these are prefixed with the word
-``profile`` because this file supports sections other than profiles::
+``region``
+    The default region to use, e.g. ``us-west-2``, ``us-west-2``, etc.
+``aws_access_key_id``
+    The access key to use.
+``aws_secret_access_key``
+    The secret access key to use.
+``aws_session_token``
+    The session token to use.  This is typically only needed when using
+    temporary credentials.  Note ``aws_security_token`` is supported for
+    backwards compatibility.
+``ca_bundle``
+    The CA bundle to use.  See the docs above on ``AWS_CA_BUNDLE`` for
+    more information.
+``metadata_service_timeout``
+    The number of seconds before timing out when retrieving data from the
+    instance metadata service.  See the docs above on
+    ``AWS_METADATA_SERVICE_TIMEOUT`` for more information.
+``metadata_service_num_attempts``
+    The number of attempts to make before giving up when retrieving data from
+    the instance metadata service.  See the docs above on
+    ``AWS_METADATA_SERVICE_NUM_ATTEMPTS`` for more information.
 
```

Comment:
Yeah I mentioned in the PR description that there's a few action items after this change, one of which is the s3 addressing config.  I mostly wanted feedback on the new layout before making any changes.

I don't really have a preference updating this PR vs. sending a new one, but my plan was to send a follow up PR that adds in the remaining docs.